public/app/features/plugins/AppRootPage.tsx

```diff
@@ -71,20 +73,30 @@ class AppRootPage extends Component<Props, State> {
     this.setState({ nav });
   };
 
+  removeStyling = (removeStyling: boolean) => {
+    this.setState({ removeStyling });
+  };
+
   render() {
     const { path, query } = this.props;
-    const { loading, plugin, nav } = this.state;
+    const { loading, plugin, nav, removeStyling } = this.state;
 
     if (plugin && !plugin.root) {
       // TODO? redirect to plugin page?
       return <div>No Root App</div>;
     }
 
     return (
-      <Page navModel={nav}>
-        <Page.Contents isLoading={loading}>
+      <Page navModel={nav} isLoading={loading}>
+        <Page.Contents isLoading={loading} removeStyling={removeStyling}>
           {plugin && plugin.root && (
-            <plugin.root meta={plugin.meta} query={query} path={path} onNavChanged={this.onNavChanged} />
+            <plugin.root
+              meta={plugin.meta}
+              query={query}
+              path={path}
+              onNavChanged={this.onNavChanged}
+              removeParentStyling={this.removeStyling}
```

Comment:
not sure why this callback exists, seems like a pageStyle property on the app is enough 

Or does the app want to change page style on dynamically? For that I think it's better to either expose seperate page components, or expose Page & PageHeader & NavModel to plugins so they can compose pages themselves. 

thoughts @ryantxu ?

Reply:
Hi @torkelo thank you for the feedback I really appreciate it, since it's all tied together I thought I will use this thread to try and explain my changes.

So the app plugin we are working on easystart requires full width no header layout, but also has other pages that have header with same layout as other plugins. 

Now what I did is: 
First I used the `removeParentStyling` to pass a boolean, whenever we wanted to remove the classes of the app which are responsible for centering the content, so by default that would always be as it was before, but only when we need the fullscreen. 

And second for the header I used the `onNavChanged` to check when no data is there to not show the header, in this case the other plugins where data is passed to header it wouldn't affect them. 

Here is a screenshot of how the wizard will look like and maybe we could add two other conditional props, one for showing or not showing the header and the other one as you suggested with enums for styling?

![wizard-intro](https://user-images.githubusercontent.com/7255779/87123881-e82f2a00-c287-11ea-875d-a13424012668.png)

Reply:
as implemented, the only way to tell the container about your settings (nav) is with callbacks.  

I hope we can revist this when we are using react router nativly

Reply:
Does it make sense to have two callbacks? e.g.

1.) `headerVisibility` which would either be `visible` or `hidden` 
2.) `pageStyles` which would have `fullWidth` `centered` 

For me to achieve the designs of the app plugin, I think these two could be enough since I can then control when I need to hide the header and when I need to have a fullscreen, but also gives me the flexibility to have the `normal` app plugin layout which we will probably need to manage the plugin.

Reply:
@torkelo @ryantxu @besartberisha instead of altering the Grafana DOM, what if Grafana had a Portal that app plugins can render to? Similar idea as we do with modals: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Modal/ModalsContext.tsx

Reply:
not sure that helps any, think exposing a Router component, and Page & Page header components makes the most long term sense:
https://github.com/grafana/grafana/pull/26245